src/lib/utils/lib.ts

```diff
@@ -195,3 +196,51 @@ export const getBlockRangesForL1Block = async (
 
   return [result[0], props.maxL2Block]
 }
+
+export async function getNativeTokenDecimals({
+  l1Provider,
+  l2Network,
+}: {
+  l1Provider: Provider
+  l2Network: L2Network
+}) {
+  const nativeTokenAddress = l2Network.nativeToken
+
+  if (!nativeTokenAddress || nativeTokenAddress === constants.AddressZero) {
+    return 18
+  }
+
+  const nativeTokenContract = ERC20__factory.connect(
+    nativeTokenAddress,
+    l1Provider
+  )
+
+  return nativeTokenContract.decimals()
```

Comment:
ERC20 don't need to necessary implement `decimals()`. If it's not implemented (so call reverts) we take that as token using 0 decimals.